### PR TITLE
[office-js, office-js-preview] Adding PDF support note for Excel on the web

### PR DESCRIPTION
**Pull request by AlexJerabek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by ElizabethSamuel-MSFT — CHANGES_REQUESTED** [review]
Waiting for feedback from engineering.

**Comment by AlexJerabek** [comment]
@ElizabethSamuel-MSFT Updated according to your feedback and that of engineering. Here's the staging site link displaying that table: https://review.docs.microsoft.com/en-us/javascript/api/office/office.document?view=word-js-preview&branch=AlexJ-Test#getfileasync-filetype--callback-

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -3450,14 +3452,14 @@ declare namespace Office {
...
-         * Excel on the web and Windows desktop: `Office.FileType.Compressed`
          * 
-         * Excel on Mac: `Office.FileType.Compressed`, `Office.FileType.Pdf`
-         *
-         * PowerPoint on the web, Windows desktop, Mac, and iPad: `Office.FileType.Compressed`, `Office.FileType.Pdf`
-         *
-         * Word on the web, Windows desktop, Mac, and iPad: `Office.FileType.Compressed`, `Office.FileType.Pdf`, `Office.FileType.Text`
+         * *Supported FileTypes, by platform*
+         *  <table>
+         *   <tr><th>                             </th><th> Windows desktop             </th><th> Office on the web           </th><th> Office on iPad      </th><th> Office on Mac               </th></tr>
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: For consistency...
```suggestion
         *   <tr><th>                             </th><th> Office on Windows             </th><th> Office on the web           </th><th> Office on iPad      </th><th> Office on Mac               </th></tr>
```

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -1843,7 +1843,9 @@ declare namespace Office {
      *
      * @remarks
      * 
-     * `FileType.Text` is only supported in Word, `FileType.Pdf` is only supported in Word on the web, Windows, and Mac, and PowerPoint.
+     * `FileType.Text` is only supported in Word.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
     * `FileType.Text` is only supported in Excel (on Windows and Mac), and Word (on Windows, Mac, and the web).
```

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -1843,7 +1843,9 @@ declare namespace Office {
...
+     * 
+     * `FileType.Pdf` is only supported in Excel (on Mac and the web), Word (on Windows, Mac, and the web), and PowerPoint.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
     * `FileType.Pdf` is only supported in Excel (on Windows, Mac, and the web), Word (on Windows, Mac, and the web), and PowerPoint.
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]
These changes will apply to both office-js and office-js-preview.

I also noticed that getFileAsync descriptions mention iOS - seems like that should be iPad instead.

**Comment by AlexJerabek** [comment]
Thanks @ElizabethSamuel-MSFT. I removed the remarks from the FileType enum, as they're another thing to keep updated. Users need to refer to the Document.getFileAsync documentation anyway.

File: types/office-js/index.d.ts [hunk 4]
```diff
@@ -3351,16 +3346,16 @@ declare namespace Office {
          * 
          * - {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/office-add-in-requirement-sets#textfile | TextFile} (when using `Office.FileType.Text`)
          *
-         * For add-ins running in Office host applications other than Office on iOS, the getFileAsync method supports getting files in slices of up 
-         * to 4194304 bytes (4 MB). For add-ins running in Office apps on iOS, the getFileAsync method supports getting files in slices of up to 
+         * For add-ins running in Office host applications other than Office on iOS, the `getFileAsync` method supports getting files in slices of up
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts]: For example, this talks about Office on iOS. I believe for WXP, it should be Office on iPad.

Comment (AlexJerabek) [reply types/office-js/index.d.ts]: Gotcha. Fixed.

Comment (ElizabethSamuel-MSFT) [reply types/office-js/index.d.ts]: Changes look good. Thx.

**Comment by orta** [comment]
Alright, thanks folks - these changes look good to me 👍 

sorry it took a bit of time for us to get around to it

**Pull request merged**